### ISSUE TITLE
Define structure for adding demo / sample projects

### ISSUE THREAD

**Issue by m-nash** [microsoft/typespec#3702]
### Clear and concise description of the problem

There is a `packages/samples` but we might want to reorganize this a bit.

Dotnet team would like to provide a sample project which demonstrates how to extend our emitter and provide your own functionality to augment what the base generation does.

For us this would include both a custom emitter and a dotnet plugin to override the generator portion.  I imagine the package structure looks similar (but much smaller) to `packages/http-client-csharp`.

### Checklist

- [X] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://typespec.io/docs/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by timotheeguerin** [comment]
Yeah be good to figure out a good design here. One option is to always have a `samples` folder for each emitter instead but this reduce discoverability I think.

If we keep samples package as the central exmaples there is a few things to think about:
- per emitter examples
- examples that apply to multiple emitter
